## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454243`
Commit: [`f2ef3f473f26dda85b438bb88652cbeb583d1043`](https://github.com/NixOS/nixpkgs/commit/f2ef3f473f26dda85b438bb88652cbeb583d1043) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/f2ef3f473f26dda85b438bb88652cbeb583d1043..pull/454243/head))
Merge: [`a73dd2661b228d2e6423712f2fd02443b1408645`](https://github.com/NixOS/nixpkgs/commit/a73dd2661b228d2e6423712f2fd02443b1408645)

Logs: https://github.com/ttrssreal/nixpkgs-review-gha/actions/runs/18769428146


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>kaitai-struct-cpp-stl-runtime</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>kaitai-struct-cpp-stl-runtime</summary>
<pre>-- The CXX compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /nix/store/z55x0q74zldi64iwamqf8wgrm2iza5rk-zlib-1.3.1/lib/libz.so (found version &quot;1.3.1&quot;)
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Success
-- Found Iconv: built in to C library
CMake Error at /nix/store/5bn5f4ivqf4xn19khh4kcg4ngnjs6spg-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /nix/store/5bn5f4ivqf4xn19khh4kcg4ngnjs6spg-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:591 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/5bn5f4ivqf4xn19khh4kcg4ngnjs6spg-cmake-4.1.2/share/cmake-4.1/Modules/FindGTest.cmake:413 (find_package_handle_standard_args)
  tests/CMakeLists.txt:3 (find_package)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>kaitai-struct-cpp-stl-runtime</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>kaitai-struct-cpp-stl-runtime</summary>
<pre>-- The CXX compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/hl19vrjrc54fvmwzp6391252j3ygnhja-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /nix/store/7wndazcsc08nn3b690byv8918xc77p16-zlib-1.3.1/lib/libz.so (found version &quot;1.3.1&quot;)
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Success
-- Found Iconv: built in to C library
CMake Error at /nix/store/wwxllnq886443diryrcxkca6b6cmn7f7-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /nix/store/wwxllnq886443diryrcxkca6b6cmn7f7-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:591 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/wwxllnq886443diryrcxkca6b6cmn7f7-cmake-4.1.2/share/cmake-4.1/Modules/FindGTest.cmake:413 (find_package_handle_standard_args)
  tests/CMakeLists.txt:3 (find_package)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>kaitai-struct-cpp-stl-runtime</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>kaitai-struct-cpp-stl-runtime</summary>
<pre>-- The CXX compiler identification is Clang 21.1.2
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/kihcn1ppp1bi8gv7hncwkaalyjx3lzdz-clang-wrapper-21.1.2/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /nix/store/yjxlzincrhac0fvc5dy2j0nbg0lqhqhp-zlib-1.3.1/lib/libz.dylib (found version &quot;1.3.1&quot;)
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Failed
-- Found Iconv: /nix/store/y4hsmfdkkr9fk51p0gk57hwvvsg32ijm-libiconv-109/lib/libiconv.dylib
CMake Error at /nix/store/06i0k56x3asszhsgwjrc1rngq0h163mx-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /nix/store/06i0k56x3asszhsgwjrc1rngq0h163mx-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:591 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/06i0k56x3asszhsgwjrc1rngq0h163mx-cmake-4.1.2/share/cmake-4.1/Modules/FindGTest.cmake:413 (find_package_handle_standard_args)
  tests/CMakeLists.txt:3 (find_package)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>kaitai-struct-cpp-stl-runtime</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>kaitai-struct-cpp-stl-runtime</summary>
<pre>-- The CXX compiler identification is Clang 21.1.2
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/xbi504848gn164lwzv2x2rclwvjxw7w1-clang-wrapper-21.1.2/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /nix/store/jyq4rjvwky8v61zi1s8j3ibzhgizccfr-zlib-1.3.1/lib/libz.dylib (found version &quot;1.3.1&quot;)
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Failed
-- Found Iconv: /nix/store/f6dybgh0x0kcp1g7vp426vb495c7mwa8-libiconv-109/lib/libiconv.dylib
CMake Error at /nix/store/g1ck4iddb7q6f5qm8pa3qqss6dp3gjlm-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /nix/store/g1ck4iddb7q6f5qm8pa3qqss6dp3gjlm-cmake-4.1.2/share/cmake-4.1/Modules/FindPackageHandleStandardArgs.cmake:591 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/g1ck4iddb7q6f5qm8pa3qqss6dp3gjlm-cmake-4.1.2/share/cmake-4.1/Modules/FindGTest.cmake:413 (find_package_handle_standard_args)
  tests/CMakeLists.txt:3 (find_package)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

--- a/pkgs/by-name/ka/kaitai-struct-cpp-stl-runtime/package.nix
+++ b/pkgs/by-name/ka/kaitai-struct-cpp-stl-runtime/package.nix
@@ -1,0 +1,73 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  gtest,
+  zlib,
+  copyPkgconfigItems,
+  makePkgconfigItem,
+  lib,
+  testers,
+  ctestCheckHook,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "kaitai-struct-cpp-stl-runtime";
+  version = "0.11";
+
+  src = fetchFromGitHub {
+    owner = "kaitai-io";
+    repo = "kaitai_struct_cpp_stl_runtime";
+    tag = finalAttrs.version;
+    sha256 = "sha256-2glGPf08bkzvnkLpQIaG2qiy/yO+bZ14hjIaCKou2vU=";
+  };
+
+  doCheck = true;
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    copyPkgconfigItems
+    ctestCheckHook
+  ];
+
+  buildInputs = [
+    zlib
+    gtest
+  ];
+
+  strictDeps = true;
+
+  # https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/issues/82
+  pkgconfigItems = [
+    (makePkgconfigItem rec {
+      name = "kaitai-struct-cpp-stl-runtime";
+      inherit (finalAttrs) version;
+      cflags = [ "-I${variables.includedir}" ];
+      libs = [
+        "-L${variables.libdir}"
+        "-lkaitai_struct_cpp_stl_runtime"
+      ];
+      variables = {
+        includedir = "${placeholder "dev"}/include";
+        libdir = "${placeholder "out"}/lib";
+      };
+      inherit (finalAttrs.meta) description;
+    })
+  ];
+
+  passthru = {
+    tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
+  };
+
+  meta = {
+    pkgConfigModules = [ "kaitai-struct-cpp-stl-runtime" ];
+    description = "Kaitai Struct C++ STL Runtime Library";
+    homepage = "https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ fzakaria ];
+  };
+})

